test_conformance/select/test_select.cpp

```diff
@@ -368,84 +373,79 @@ static int doTest(cl_command_queue queue, cl_context context, Type stype, Type c
         cmp_stride = block_elements * step * (0xffffffffffffffffULL / 0x100000000ULL + 1);
 
     log_info("Testing...");
-    d = init_genrand( gRandomSeed );
-    uint64_t i;
-    for (i=0; i < blocks; i+=step)
+    d0 = init_genrand( gRandomSeed );
+    randBuf = clCreateBuffer( context, CL_MEM_READ_ONLY, BUFFER_SIZE, nullptr, &err );
+    if( err ) { log_error( "Error: could not allocate src1 buffer\n" );  ++s_test_fail; goto exit; }
+    initSrcBuffer(randBuf, stype, d0);
+    s1 = clEnqueueMapBuffer( queue, src1, CL_TRUE, CL_MAP_WRITE, 0, BUFFER_SIZE, 0, nullptr, nullptr, &err );
+    if( err ){ log_error( "Error: Could not map src1" ); goto exit; }
+    s2 = clEnqueueMapBuffer( queue, src2, CL_TRUE, CL_MAP_WRITE, 0, BUFFER_SIZE, 0, nullptr, nullptr, &err );
+    if( err ){ log_error( "Error: Could not map src2" ); goto exit; }
+    s3 = clEnqueueMapBuffer( queue, cmp, CL_TRUE, CL_MAP_WRITE, 0, BUFFER_SIZE, 0, nullptr, nullptr, &err );
+    if( err ){ log_error( "Error: Could not map cmp" ); goto exit; }
+    // Set up the input data to change for each block
+    clEnqueueCopyBuffer(queue, randBuf, src1, 0, 0, BUFFER_SIZE, 0, nullptr, nullptr);
+    // Set up the input data to change for each block
+    clEnqueueCopyBuffer(queue, randBuf, src2, 0, 0, BUFFER_SIZE, 0, nullptr, nullptr);
```

Comment:
Is the intent here to initialize both input buffers to the same data. If so, this would not be good practice. The original intent of the code was to have 2  independently randomized inputs.